### PR TITLE
fixes bug 1312537 - Track version number in crash_reports.json JSON S…

### PR DESCRIPTION
**Pull request by peterbe**
…chema

CC @mreid-moz

### CODE REVIEW

File: socorro/schemas/README.md [hunk 1]
```diff
@@ -1,5 +1,46 @@
...
+
+The JSON Schema should contain a key called `$target_version`.
+
+* It's important this is an integer that goes up
+* *Don't change this* if you're...
+  * Adding more keys
+  * Editing comments (content of `description` values)
+* *Do change this* if you're...
+  * Changing the type definition of an *existing* key in any way
+  * Add or remove keys from a `required` sub-key
```
Comment (adngdb) [inline socorro/schemas/README.md]: This is unclear. If I add a new key to the ``json_dump`` object, do I increment the version number?

File: socorro/schemas/README.md [hunk 2]
```diff
@@ -1,5 +1,46 @@
...
+Yes, if you add more keys, don't change the version.
+For example, if you want to add...:
+
+```json
+"addons_checksum": {
+    "type": ["string", "null"],
+    "description": "Sample specimen"
+}
+```
+...to the root or inside an existing key. Then **don't** change the version.
```
Comment (adngdb) [inline socorro/schemas/README.md]: s/key. Then/key, then

File: socorro/schemas/README.md [hunk 3]
```diff
@@ -1,5 +1,46 @@
...
+
+```diff
+"addons_checksum": {
+-   "type": ["string", "null"],
++   "type": ["integer", "null"],
+    "description": "Sample specimen"
+},
+```
+Then you have to change the `$target_version` number to `$target_version+1`
+whatever that new number might be.
```
Comment (adngdb) [inline socorro/schemas/README.md]: I think you can simplify that by saying ``Then you have to increment the `$target_version` number. ``.

**Comment by peterbe** [comment]
@adngdb Better?

**Comment by adngdb** [comment]
Yup!

**Pull request merged**